bug: gregorian_time.c fails to compile

:bug: Your bug may already be reported!
Please search on the [issue tracker](https://github.com/NCAR/DART/issues) before creating a new issue.

The utility gregorian_time.c is in assimilation_code/modules/utilities

### Describe the bug

1. List the steps someone needs to take to reproduce the bug.  
    Compile gregorian_time.c  
2. What was the expected outcome?
     Successful compilation
3. What actually happened?  
     fails with compilers enforcing ICO C99

### Error Message  
```
$ icx gregorian_time.c 
gregorian_time.c:40:1: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
   40 | main(int argc, char **argv)
      | ^
      | int
1 error generated.
```

### Which model(s) are you working with?
none

### Version of DART      
v11.14.2

### Have you modified the DART code?   
No  
If your code changes are available on GitHub, please provide the repository.

### Build information
Please describe:  
 1. Derecho
 2. intel icx
     icx --version
Intel(R) oneAPI DPC++/C++ Compiler 2024.2.1 (2024.2.1.20240711)


Fix:

```diff
-main(int argc, char **argv)
+int main(int argc, char **argv)
```

--- a/assimilation_code/modules/utilities/gregorian_time.c
+++ b/assimilation_code/modules/utilities/gregorian_time.c
@@ -37,7 +37,7 @@ void convert(int direction, int *days, int *secs,
 enum err vet_time(int days, int secs);
 enum err vet_date(int year, int month, int day, int hour, int min, int sec);
 
-main(int argc, char **argv)
+int main(int argc, char **argv)
 {
     int days, secs, year, month, day, hour, min, sec;
     enum err error;

